lib/fa2/asset/single_asset.impl.jsligo

```diff
@@ -0,0 +1,248 @@
+      const sender_ = Tezos.get_sender();
+      if (sender_ == from_) return unit else {
+         const authorized =
+            match(Big_map.find_opt(from_, operators)) {
+               when (Some(a)):
+                  a
+               when (None()):
+                  Set.empty
+            };
+         if (Set.mem(sender_, authorized)) return unit else failwith(
+            Errors.not_operator
+         )
+      }
+   };
+   const assert_update_permission = (owner: address): unit => {
+      return assert_with_error(
+         owner == Tezos.get_sender(),
+         "The sender can only manage operators for his own token"
+      )
+   };
+   export const add_operator = (
+      operators: Operators,
+      owner: address,
+      operator: operator
+   ): Operators => {
+      if (owner == operator) return operators else {
+         const _ = assert_update_permission(owner);
+         let auths =
+            match(Big_map.find_opt(owner, operators)) {
+               when (Some(os)):
+                  os
+               when (None()):
+                  Set.empty
+            };
+         auths = Set.add(operator, auths);
+         return Big_map.update(owner, Some(auths), operators)
+      }
+   };
+   export const remove_operator = (
+      operators: Operators,
+      owner: address,
+      operator: operator
+   ): Operators => {
+      if (owner == operator) return operators else {
+         const _ = assert_update_permission(owner);
+         const auths =
+            match(Big_map.find_opt(owner, operators)) {
+               when (None()):
+                  None()
+               when (Some(ops)):
+                  do {
+                     let os = Set.remove(operator, ops);
+                     if (Set.size(os) == 0n) return None() else return Some(os)
+                  }
+            };
+         return Big_map.update(owner, auths, operators)
+      }
+   }
+   //export namespace Ledger {
+
+   export const get_for_user = (ledger: Ledger, owner: address): nat =>
+      match(Big_map.find_opt(owner, ledger)) {
+         when (Some(tokens)):
+            tokens
+         when (None()):
+            0 as nat
+      };
+   const update_for_user = (ledger: Ledger, owner: address, amount_: nat): Ledger =>
+      Big_map.update(owner, Some(amount_), ledger);
+   export const decrease_token_amount_for_user = (
+      ledger: Ledger,
+      from_: address,
+      amount_: nat
+   ): Ledger => {
+      let tokens = get_for_user(ledger, from_);
+      const _ = assert_with_error(tokens >= amount_, Errors.ins_balance);
+      tokens = abs(tokens - amount_);
+      return update_for_user(ledger, from_, tokens)
+   };
+   export const increase_token_amount_for_user = (
+      ledger: Ledger,
+      to_: address,
+      amount_: nat
+   ): Ledger => {
+      let tokens = get_for_user(ledger, to_);
+      tokens = tokens + amount_;
+      return update_for_user(ledger, to_, tokens)
+   }
+   //};
+   //export namespace Storage {
```

Comment:
File should be clean, code comments should be removed

Reply:
I will clean legacy code